src/sentry/models/release.py

```diff
@@ -768,14 +774,8 @@ def set_commits(self, commit_list):
                         key=data["id"],
                         defaults=commit_data,
                     )
-                    if not created:
-                        commit_data = {
-                            key: value
-                            for key, value in commit_data.items()
-                            if getattr(commit, key) != value
-                        }
-                        if commit_data:
-                            commit.update(**commit_data)
```

Comment:
this is ever-so-slightly slower in that it sometimes updates more things -- but greatly simplifies the types here (`commit_data` being reassigned to `dict[str, object]` here)